cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -120,12 +122,13 @@
   private Transaction transaction;
   private Runnable cleanupTask;
   private volatile boolean stopRequested;
+  private final Store store;
 
   MapReduceRuntimeService(CConfiguration cConf, Configuration hConf,
                           MapReduce mapReduce, MapReduceSpecification specification,
                           DynamicMapReduceContext context,
                           Location programJarLocation, LocationFactory locationFactory,
-                          StreamAdmin streamAdmin, TransactionSystemClient txClient) {
+                          StreamAdmin streamAdmin, TransactionSystemClient txClient, Store store) {
```

Comment:
You shouldn't need to pass the store to this class. The `MapReduceRuntimeService` is a guava Service, hence you can attach a service listener in the program runner, before starting this service and do the writing to store in the listener.